File: mobile/lib/core/services/esp32_service.dart
import '../constants/app_constants.dart';
import 'api_client.dart';

class ESP32Service {
  final ApiClient _apiClient = ApiClient();

  // Get ESP32 status
  Future<Map<String, dynamic>> getESP32Status() async {
    try {
      final response = await _apiClient.get(AppConstants.esp32StatusEndpoint);

      if (response.statusCode == 200 && response.data['success'] == true) {
        return {
          'success': true,
          'data': response.data['data'],
        };
      }

      return {
        'success': false,
        'message':
            response.data['message'] ?? 'Không thể tải trạng thái ESP32',
      };
    } catch (e) {
      return {'success': false, 'message': 'Lỗi: ${e.toString()}'};
    }
  }
}

File: frontend/src/services/esp32.service.ts
import apiClient from "./api.client";
import { API_ENDPOINTS } from "../config/api.config";

export interface ESP32StatusResponse {
  deviceId: string;
  status: "online" | "offline";
  isOnline: boolean;
  lastSeen: string;
  lastSeenSeconds: number;
  ipAddress: string | null;
  updatedAt: string;
}

export const esp32Service = {
  async getStatus(): Promise<ESP32StatusResponse> {
    const response = await apiClient.get<{
      success: boolean;
      data: ESP32StatusResponse;
    }>(API_ENDPOINTS.ESP32.STATUS);
    return response.data.data;
  },
};
